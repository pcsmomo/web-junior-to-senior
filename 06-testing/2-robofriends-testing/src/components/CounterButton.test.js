import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('renders without crashing', () => {
  expect(shallow(<CounterButton />)).toMatchSnapshot();
});

it('correnctly increments the counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />);
  
  wrapper.find('[id="counter"]').simulate('click');
  expect((wrapper.state())).toEqual({count: 1})
  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect((wrapper.state())).toEqual({count: 3})
  wrapper.find('[id="counter"]').simulate('keypress'); // there no keypress event
  expect(wrapper.state()).toEqual({ count: 3 });
  expect(wrapper.props().color).toEqual('red');
});