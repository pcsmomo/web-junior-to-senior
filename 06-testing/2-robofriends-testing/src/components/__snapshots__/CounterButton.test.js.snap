// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
